### PR TITLE
Implement agency dashboard for small screen view

### PR DESCRIPTION
**Pull request by yashwin**
#### Changes proposed in this Pull Request

* This PR includes the implementation of a site view for the agency dashboard which includes a card view for the small screens (<960px)

#### Testing instructions

**Prerequisites**

Since this change is made specifically for agencies, you will have to set yourself(partner) as an agency - 2c49b-pb

**NOTE**

- We have a couple of things pending which will be added in the follow-up PR - handling track events, pagination, and truncating long site names
- The sites are not real sites, so the links will not work as expected, just making sure the links are taking the user to the proper links will do. The site names and row values may not be the same as in the screenshot.
- The page title and subtitle will be added in the follow-up PR.
- Actions for the table will be added from this PR - https://github.com/Automattic/wp-calypso/pull/63594

**Instructions**

1. Run `git checkout add/implement-agency-dashboard-table` and `yarn start-jetpack-cloud`
2. Apply D79008-code for the mock API to work.
3. Visit http://jetpack.cloud.localhost:3000/, and you'll be redirected to http://jetpack.cloud.localhost:3000/dashboard - make sure the table UI is loaded as shown in the screenshot.
4. We are currently hitting the mock API, so the values displayed are not accurate, but the row value formatting works as expected, make sure the actions are working as expected and rows are formatted accordingly as shown in the screenshot.
5. Make sure the UI is looking fine in all the device resolutions.
6. The UI shown on the large screen(>961px) will be a table view and a card view should be shown on smaller devices(<961px)
7. On devices <960, the card should expand/collapse when clicked on the card header except for the actions on the right.

Row values - 1202076982646589-as-1202122273254270

- `+ Add` should take the user to the issue license page(backup/scan).
- All the actions should be disabled if the site is down as shown in the screenshot.
- Backup/Scan when warning/failed/active(check/tick icon)/progress should take the user to the backup/scan site page when clicked. 
- Monitor when active(check/tick icon) should show a tooltip that says `Monitor is on and your site is online`
- Should show `Failed`, `Warning` in Backup/Scan when the backup is failed or has a warning.
- Should show `# Threats` in Scan when a site has threats where # is the no of threats.
- Should show `Site Down` in Monitor when the site is down and should take the user to JP debugger when clicked on it.
- Should show `# Available` in Plugins when the site has available updates and should take to [wordpress -> updates](https://wordpress.com/plugins/updates/:site) and show active when no updates are available
- Monitor will show active(check/tick icon) when it is working as expected and `-` when it is not enabled.
- `Fix Now` should redirect the user to `/settings/disconnect-site/${ siteUrl }?type=down` 

Screenshots

Dashboard View(Large Screen)

Screenshot 2022-05-11 at 6 25 11 PM

Mobile view - 320px

Screenshot 2022-05-13 at 12 13 19 PM

Tablet view - 768px

Screenshot 2022-05-13 at 12 13 33 PM

Related to 1202076982646589-as-1202122273254256

### CODE REVIEW

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-card/index.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	const toggleContent = isExpanded ? (
+		<Gridicon icon="chevron-up" className="site-card__vertical-align-middle" size={ 18 } />
+	) : (
+		<Gridicon icon="chevron-down" className="site-card__vertical-align-middle" size={ 18 } />
+	);
+
+	const rowKeys = Object.keys( rows );
+
+	const headerItem = rowKeys[ 0 ];
+	const expandedContentItems = rowKeys.filter( ( row, index ) => index > 0 );
```
Comment (manzoorwanijk) [inline client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-card/index.tsx]: Why not this?
```suggestion
	const [ headerItem, ...expandedContentItems ] = Object.keys( rows );
```

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-card/index.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+		[ key: string ]: any;
+	};
+}
+
+export default function SiteCard( { rows, columns }: Props ): ReactElement {
+	const [ isExpanded, setIsExpanded ] = useState( false );
+
+	const toggleIsExpanded = () => {
+		setIsExpanded( ! isExpanded );
+	};
```
Comment (manzoorwanijk) [inline client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-card/index.tsx]: If new state depends upon previous state, it's better to use [Functional Updates](https://reactjs.org/docs/hooks-reference.html#functional-updates).

```suggestion
	const toggleIsExpanded = () => {
		setIsExpanded( ( prevValue ) => ! prevValue );
	};
```

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-card/index.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+	const toggleContent = isExpanded ? (
+		<Gridicon icon="chevron-up" className="site-card__vertical-align-middle" size={ 18 } />
+	) : (
+		<Gridicon icon="chevron-down" className="site-card__vertical-align-middle" size={ 18 } />
+	);
+
+	const rowKeys = Object.keys( rows );
```
Comment (manzoorwanijk) [inline client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-card/index.tsx]: I am little bit confused here. `rows` is an object with keys like `site`, `scan`, `plugin` etc. and `Object.keys( rows )` would return an array like `[ 'site', 'scan', 'plugin' ]`, but do we expect the array to be in certain order? `headerItem` below should be `'site'`. Is the sorting of the array guaranteed here?

Some unit tests may confirm our assumption.

Comment (yashwin) [reply client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-card/index.tsx]: Very good catch, @manzoorwanijk. I have made the changes accordingly.

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-content/index.tsx [hunk 4]
```diff
@@ -13,20 +15,50 @@ export default function SiteContent(): ReactElement {
...
 		<>
 			<SiteTable isFetching={ isLoading } columns={ columns } items={ sites } />
+			<div className="site-content__mobile-view">
+				<>
+					{ isLoading || error ? (
+						<JetpackLogo size={ 72 } className="site-content__logo" />
+					) : (
+						<>
+							{ sites.length > 0 &&
+								sites.map( ( rows: any, index: number ) => (
```
Comment (manzoorwanijk) [inline client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-content/index.tsx]: You don't need to type these arguments for `.map` callback. TS sets types for those automagically.
```suggestion
								sites.map( ( rows, index ) => (
```

**Review by manzoorwanijk — APPROVED** [review]
Works well for me.

Screenshot 2022-05-17 at 11 20 31 AM


My only suggestion would be to improve TS types for the components and the utility function like `getRowMetaData`, `formatSites` etc.
For example the `Props` defined in `SiteTable` could be extracted and re-used in `SiteCard`.

**Comment by yashwin** [comment]
Thanks for the quick review and feedback, @manzoorwanijk. I have moved the types to the common file and reusing them now in the components.

**Review by manzoorwanijk — APPROVED** [review]
Cool. Thank you for those changes.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7313726

Thank you @yashwin for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7313726).

**Pull request merged**